state/protocol/badger/snapshot.go

```diff
@@ -284,11 +284,58 @@ func (s *Snapshot) SealingSegment() (*flow.SealingSegment, error) {
 		return nil
 	}
 
-	err = fork.TraverseForward(s.state.headers, s.blockID, scraper, fork.IncludingBlock(seal.BlockID))
+	lowestBlockID := seal.BlockID
+	err = fork.TraverseForward(s.state.headers, s.blockID, scraper, fork.IncludingBlock(lowestBlockID))
 	if err != nil {
 		return nil, fmt.Errorf("could not traverse sealing segment: %w", err)
 	}
 
+	// strictly speaking sealing segment length should be bounded by next formula:
+	// limitHeight = max(s.block.Height - maxSealingSegmentLength, SporkRootBlockHeight)
+
+	highestBlock, err := s.state.headers.ByBlockID(s.blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get highest block in sealing segment: %w", err)
+	}
+
+	// start by assuming that we won't go lower than spork root block height
+	limitHeight := s.state.sporkRootBlockHeight
+	sealingSegmentMaxLength := s.state.maxSealingSegmentLength
+	if highestBlock.Height > sealingSegmentMaxLength {
+		lowestSealingSegmentHeight := highestBlock.Height - sealingSegmentMaxLength
+		if limitHeight < lowestSealingSegmentHeight {
+			limitHeight = lowestSealingSegmentHeight
+		}
+	}
+
+	lowestSealingSegmentBlock, err := s.state.headers.ByBlockID(lowestBlockID)
```

Comment:
```suggestion
	// the highest sealed block is the first block above any extra blocks
	highestSealedBlock, err := s.state.headers.ByBlockID(lowestBlockID)
```
Similar to my comment about `SealingSegment.Lowest()`, I think a different name is more appropriate for this block now.